modules/cats/src/smithy4s/interopcats/SchemaVisitorShow.scala

```diff
@@ -0,0 +1,150 @@
+package smithy4s.interopcats
+
+import cats.Show
+import cats.implicits.toContravariantOps
+import smithy4s._
+import smithy4s.capability.EncoderK
+import smithy4s.interopcats.instances.ShowInstances._
+import smithy4s.schema.{
+  Alt,
+  CollectionTag,
+  EnumValue,
+  Field,
+  Primitive,
+  Schema,
+  SchemaAlt,
+  SchemaField,
+  SchemaVisitor
+}
+import smithy4s.schema.Alt.Precompiler
+
+object SchemaVisitorShow extends SchemaVisitor[Show] { self =>
+
+  override def primitive[P](
+      shapeId: ShapeId,
+      hints: Hints,
+      tag: Primitive[P]
+  ): Show[P] = primShowPf(tag)
+
+  override def collection[C[`2`], A](
+      shapeId: ShapeId,
+      hints: Hints,
+      tag: CollectionTag[C],
+      member: Schema[A]
+  ): Show[C[A]] = {
+    implicit val showSchemaA: Show[A] = self(member)
+    tag match {
+      case CollectionTag.ListTag => Show[List[A]]
+
+      case CollectionTag.SetTag => Show[Set[A]]
+
+      case CollectionTag.VectorTag => Show[Vector[A]]
+
+      case CollectionTag.IndexedSeqTag => Show[Seq[A]].contramap(_.toIndexedSeq)
+    }
+  }
+
+  override def union[U](
+      shapeId: ShapeId,
+      hints: Hints,
+      alternatives: Vector[SchemaAlt[U, _]],
+      dispatch: Alt.Dispatcher[Schema, U]
+  ): Show[U] = {
+
+    val precomputed: Precompiler[Schema, Show] = new Precompiler[Schema, Show] {
+      override def apply[A](label: String, instance: Schema[A]): Show[A] = {
+        self(instance)
```

Comment:
Echoing my comment on the struct implementation, I think we could do better here, and have the `Show` for unions be similar to the ones of structures, by rendering the name of the `ShapeId` of the union, as well as the label. 

So the Show for `IntOrInt` would render something like : 

``` 
IntOrInt(intValue0 = 0) 
```

or 

```
IntOrInt(intValue1 = 1) 
```